examples/std_global_allocator.rs

```diff
@@ -7,14 +7,15 @@ use talc::*;
 static mut START_ARENA: [u8; 10000] = [0; 10000];
 
 #[global_allocator]
-// the mutex provided by the `spin` crate is used here as it's a sensible choice
+// The mutex provided by the `spin` crate is used here as it's a sensible choice
 static ALLOCATOR: Talck<spin::Mutex<()>, ClaimOnOom> =
-    // we need to use the ClaimOnOom OOM handler or similar as allocations may
-    // occur prior to invocation of the program entrypoint main(), so claiming some
-    // memory must be done on-demand
+    // An OOM handler such as ClaimOnOom is required, as allocations
+    // may occur prior to invoking the program entrypoint `main()`,
+    // thus needs to support claiming memory on demand
```

Comment:
```suggestion
    // Allocations may occur prior to the execution of `main()`, thus support for 
    // claiming memory on-demand is required, such as ClaimOnOom.
```
I think this is more clear. Let me know if it meets your goals.